src/renameAll.js

```diff
@@ -2,22 +2,33 @@ const curry = require('ramda/src/curry')
 
 // renameAll :: { k: v } -> { k: v } -> { k: v }
 const renameAll = (renames, obj) => {
-  obj = Object.assign({}, obj)
+  const newObj = {}
+  const tosMap = {}
+
+  for (let k in obj) {
+    newObj[k] = obj[k]
+  }
```

Comment:
One less thing to polyfill, IMO, but I don't think we're willing to die on this object copying hill

Reply:
Why are we iterating twice anyway? Can we get by with iterating over `obj` once?

I just did this elsewhere. Is just for `rename`, but same principle: https://github.com/articulate/authentic/pull/21/files#diff-27a1c86133de4310373decfdad200a91R1

I think is may also solve this problem if you don't have to track state of what we've already touched.

```js
const newObj = {}
for (let prevKey in obj) {
  const nextKey = renames[prevKey] || prevKey
  nextObj[nextKey] = obj[prevKey]
}
```

Reply:
It keeps it as close to the original implementation as possible, but if this is a better idea, then ✅ . @jesspoemape want to give it a go?

Reply:
(This is that hill I was talking about not being willing to die on.)

Short of refactoring how you suggest (up to Jessica if she wants to), the alternative would be to take out the loop and put back in the `Object.assign` so that the only added complexity is the conditional to determine "do you delete the key or not?"

Reply:
@rpearce @mgreystone 
Just pushed up a refactor as per Markus' suggestion

Reply:
That's pretty slick